include/beman/inplace_vector/inplace_vector.hpp

```diff
@@ -252,6 +252,9 @@ prospectively choose to deem waived or otherwise exclude such Section(s) of
 the License, but only in their entirety and only with respect to the Combined
 Software.
  */
+
+#include <beman/inplace_vector/config.hpp>
```

Comment:
If this dose not go agains Beman guidelines, I would like to have the header only library independant of cmake generated config files.

```suggestion
#if !defined(__has_include) || __has_include(<beman/inplace_vector/config.hpp>)
#include <beman/inplace_vector/config.hpp>
#endif
```

` !defined(__has_include)`  fallback in case compiler does not have this feature

Reply:
> would like to have the header only library independant of cmake generated config files.

I'm not quite sure what you're saying?  Is it just that you'd like to be able to lift the inplace_vector.hpp outside of this tree and drop into a project and have it *just work*? If yes, there's certainly no guideline against that.

And sorry, is the config.hpp generated?  If it is that should be reconsidered...

Reply:
yes, I'd like to be able to use the library without cmake, but according to [CPP.NO_FLAG_FORKING](https://github.com/bemanproject/beman/blob/main/docs/BEMAN_STANDARD.md#c)  it is required to use cmake options and generate config.hpp. 

I get why the rule was added and don’t want to be fighting windmills, but in my opinion, it’s too strict (it should be a recommendation instead of a requirement), at least for a header-only library

Reply:
yeah, and actually I was somehow not paying attention when that requirement was added -- I don't think I agree with it.  I think I understand what @camio is going trying to prevent, but I don't think that generating a config actually solves the problem in the header only cases.  Many many developers will tend to pull the .hpp file and just put it in their project...

Reply:
@JeffGarland, building consensus for that rule was grueling. If you want to relitigate, let's do it in discourse instead of on this PR. The PR should follow the standard as-is and if the standard changes, we can always make future modifications.

Folks who want to drop headers into their project can also drop a generated `config.hpp`. I don't think that's too much to ask.